File: plugin/src/main/java/com/github/weisj/darkmode/AutoDarkModeStartupListener.java
package com.github.weisj.darkmode;

import com.intellij.ide.AppLifecycleListener;
import com.intellij.openapi.components.ServiceManager;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Optional;

public class AutoDarkModeStartupListener implements AppLifecycleListener {

    @Override
    public void appFrameCreated(@NotNull List<String> commandLineArgs) {
        ServiceManager.getService(AutoDarkMode.class).start();
    }

    @Override
    public void appClosing() {
        Optional.ofNullable(ServiceManager.getServiceIfCreated(AutoDarkMode.class)).ifPresent(AutoDarkMode::stop);
    }
}

File: plugin/src/main/java/com/github/weisj/darkmode/AutoDarkModeStartupListener.kt
/*
 * MIT License
 *
 * Copyright (c) 2020 Jannis Weis
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */
package com.github.weisj.darkmode

import com.intellij.ide.AppLifecycleListener
import com.intellij.openapi.application.ApplicationManager

class AutoDarkModeStartupListener : AppLifecycleListener {
    override fun appFrameCreated(commandLineArgs: List<String>) {
        IntellijNotificationService.initializeIfNeeded()
        ApplicationManager.getApplication().getService(AutoDarkMode::class.java).start()
    }

    override fun appClosing() {
        ApplicationManager.getApplication().getServiceIfCreated(AutoDarkMode::class.java)?.stop()
    }
}
